fix(BDM): Update the BDM doc to include new API methods (#1579)
bonitasoft/bonita-doc: modules/ROOT/pages/how-a-bdm-is-deployed.adoc

The `TenantAdministrationAPI` exposes several methods:

* https://javadoc.bonitasoft.com/api/{javadocVersion}/org/bonitasoft/engine/api/TenantAdministrationAPI.html#installBusinessDataModel-byte:A-[installBusinessDataModel()] that installs a new BDM. Fails if one already exists. Call uninstallBusinessDataModel()

> first if a BDM already exists. Deprecated in 2021.2, as UpdateBusinessDataModel() covers the same use case, and should be used instead.

Reviewer: ```suggestion
first if a BDM already exists. Deprecated in 2021.2, as updateBusinessDataModel() covers the same use case, and should be used instead.
```

> * https://javadoc.bonitasoft.com/api/{javadocVersion}/org/bonitasoft/engine/api/TenantAdministrationAPI.html#updateBusinessDataModel--[updateBusinessDataModel()] Updates the BDM. Uninstalls the old BDM if there was one and installs the new BDM.

Reviewer: ```suggestion
* https://javadoc.bonitasoft.com/api/{javadocVersion}/org/bonitasoft/engine/api/TenantAdministrationAPI.html#updateBusinessDataModel--[updateBusinessDataModel()] that updates the BDM. It uninstalls the previous BDM if applicable, and installs the new one.
```

(and from all the process classloaders of the tenant). *Does not* update or remove anything from the database.
* https://javadoc.bonitasoft.com/api/{javadocVersion}/org/bonitasoft/engine/api/TenantAdministrationAPI.html#cleanAndUninstallBusinessDataModel--[cleanAndUninstallBusinessDataModel()]: same as above plus
*drops all the BDM tables* resulting in a complete data loss. *Not for production*. Use with caution.

> * https://javadoc.bonitasoft.com/api/{javadocVersion}/org/bonitasoft/engine/api/TenantAdministrationAPI.html#updateBusinessDataModel--[updateBusinessDataModel()] Updates the BDM. Uninstalls the old BDM if there was one and installs the new BDM.

Reviewer 2: If I am not wrong it cover as well the case that there is no BDM (fresh install). Should we update the installBDM description?

Author: Will do. That being said, the install method still exists in the API, and the limitations on it still apply.

Author: Honestly, I feel we should depreciate the install method.

* you must be logged in as the tenant administrator
* go to the BPM services page, and pause the services
* go to the Business Data Model page, select the BDM zip file to deploy
* the portal will call the method updateBusinessDataModel() to install the new BDM.
This method implicitly performs the following steps in a single database transaction:

> ** if a BDM already exists on the current tenant, uninstallBusinessDataModel() is called.
> ** then the portal calls installBusinessDataModel() on the Engine API, which generates the Java model, loads it in the tenant

Reviewer: ```suggestion
** if a BDM already exists on the current tenant, it is uninstalled
** the new BDM is installed / deployed
** the Bonita Engine loads the new BDM in the tenant
```

* you must be logged in as the tenant administrator
* go to the BPM services page, and pause the services
* go to the Business Data Model page, select the BDM zip file to deploy
* the portal will call the method updateBusinessDataModel() to install the new BDM.

> This method implicitly calls the two following methods in a single database transaction:

Reviewer 2: ```suggestion
This method implicitly perform the following steps in a single database transaction:
```

Reviewer 2: I am not sure about the "single database transaction"
why database?

Reviewer: ~perform~ => perform**s**

Reviewer: ```suggestion
This method implicitly performs the following steps in a single database transaction:
```
